docs/how-to-integrate-a-frontend.md

```diff
@@ -0,0 +1,247 @@
+---
+sidebar_position: 8
+---
+
+# How to integrate Smart Contract With a Frontend
+
+In this part, we will create a full-stack decentralized application (dApp) by connecting the [Tic-Tac-Toe](./tutorials/tic-tac-toe.md) smart contract to a frontend project and interacting with it.
+
+For this dApp, we will use [React](https://reactjs.org/) as our frontend framework. We assume that you already have the basic knowledge of frontend development, so we will not spend much time introducing this part of the code, but mostly be focusing on how to interact with the smart contract in the frontend project.
+
+:::note
+You can use any frontend framework to build the dApp.
+:::
+
+## Setup
+
+### Scaffold
+
+It's free to run the following command to create a React app from the very beginning.
+
+```bash
+npx create-react-app tic-tac-toe --template typescript
+```
+
+Or directly clone the [repository](https://github.com/sCrypt-Inc/tic-tac-toe) that we have created. The `onlyweb` branch of this repository only contains the frontend code.
+
+```bash
+git clone -b onlyweb https://github.com/sCrypt-Inc/tic-tac-toe
+```
+![](../static/img/clone.png)
+
+Later, we will do most work under the `src` directory.
+
+### Install scryptTS SDK
+
+The sCrypt SDK - [scryptTS](https://scrypt.io/scrypt-ts) - enables you to easily compile, test, deploy, and call contracts.
+
+Step into the project directory, then use the `scrypt-cli` command line tool to install the SDK.
+
+```bash
+cd tic-tac-toe
+npx scrypt-cli init
+```
+
+This `init` command will also install the dependencies and initialize the contract development environment.
+
+After this initialization, we are ready to go!
+
+## Load Contract
+
+Before interacting with the contract in the front end, we need to load it first. 
+
+When a contract source file is compiled, it will deliver a JSON file, we call it the contract artifact.
+
+Generally, as a frontend developer, you will get this contract artifact from your backend partner or contract developer, then load it in the front end, and that's all.
+
+We'll take a look at how to generate the artifact by ourselves first.
+
+### Compile Contract
+
+When you have the contract source file, copy it into `src/contracts` directory.
+
+![](../static/img/copy-source-file.png)
+
+Then run the following command to compile the contract.
+
+```bash
+npx scrypt-cli compile
+```
+
+![](../static/img/compile.png)
+
+After the compilation, you will get the artifact in `artifacts/src/contract` directory.
+
+![](../static/img/artifacts.png)
+
+### Load Artifact
+
+Now with the artifact file `ticTacToe.json`, we directly load it in the `index.tsx` file.
+
+```ts
+import { TicTacToe } from './contracts/ticTacToe';
+var artifact = require('../artifacts/src/contracts/ticTacToe.json');
+TicTacToe.loadArtifact(artifact);
+```
+
+## Integrate Wallet
+
+We will integrate [Sensilet](https://sensilet.com/), a MetaMask-like wallet, into our project.
```

Comment:
Add :::info similar to https://docs.minaprotocol.com/zkapps/how-to-write-a-zkapp-ui#connecting-your-zkapp-with-a-users-wallet, saying other wallets can be added by following https://scrypt.io/scrypt-ts/how-to-add-a-signer.

Reply:
@gitzhou This is yet to be done.